### PR TITLE
Fix TestEnvironmentOfChildProcess to handle newlines

### LINKED ISSUE #3237: TestEnvironmentOfChildProcess fails (intermittently?)
It looks like it's pulling in the PR description and something is going wrong with the roundtripping thereof:

http://dotnet-ci.cloudapp.net/job/dotnet_corefx_prtest/4024/

```
Assert+WrapperXunitException : File path: d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Diagnostics.Process\tests\ProcessStartInfoTests.cs. Line: 231
----
Expected: ghprbPullLongDescription=Since CryptographicException has no constructor that takes both an HResult and a Message this is being achieved by a private derived type which can set both properties., , Fixes #3232.
Actual: ghprbPullLongDescription=Since CryptographicException has no constructor that takes both an HResult and a Message this is being achieved by a private derived type which can set both properties. Fixes #3232.
Expected: True
Actual: False
+++++++++++++++++++
STACK TRACE:
at Assert.WrapException(Exception e, String callerFilePath, Int32 callerLineNumber)
  in d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\Common\tests\System\Diagnostics\AssertWithCallerAttributes.cs:line 582
at Assert.True(Boolean condition, String userMessage, String path, Int32 line)
  in d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\Common\tests\System\Diagnostics\AssertWithCallerAttributes.cs:line 577
at System.Diagnostics.ProcessTests.ProcessStartInfoTests.TestEnvironmentOfChildProcess()
  in d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Diagnostics.Process\tests\ProcessStartInfoTests.cs:line 231
----- Inner Stack Trace -----
at Assert.True(Boolean condition, String userMessage, String path, Int32 line)
  in d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\Common\tests\System\Diagnostics\AssertWithCallerAttributes.cs:line 574
```

### PR DESCRIPTION
**Pull request by stephentoub**
Something changed recently in the CI environment, such that Jenkins is now putting PR/commit/comment information in environment variables.  This is breaking the TestEnvironmentOfChildProcess test, which assumes (poorly) environment variables won't have newlines in them, as it's using newlines for parsing.  This fix basically just switches to using something other than a newline for parsing.  It also includes a few other simplifications.

Fixes #3237

### CODE REVIEW

**Comment by stephentoub** [comment]
cc: @Priya91, @Maxwe11, @akoeplinger

**Comment by Priya91** [comment]
LGTM

File: src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs [hunk 1]
```diff
@@ -185,70 +185,52 @@ public void TestEnvironmentProperty()
...
 
-                    Assert.True(false, string.Format("Expected: {0}{1}Actual: {2}", expected, Environment.NewLine, actual));
+                // Validate against StartInfo.Environment.
+                var startInfoEnv = new HashSet<string>(p.StartInfo.Environment.Select(e => e.Key + "=" + e.Value));
+                if (!startInfoEnv.SetEquals(actualEnv))
+                {
+                    Assert.True(false, string.Format("Expected: {0}{1}Actual: {2}",
+                        string.Join(", ", startInfoEnv.Except(actualEnv)), 
+                        Environment.NewLine,
+                        string.Join(", ", actualEnv.Except(startInfoEnv))));
```
Comment (akoeplinger) [inline src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs]: `Assert.Equal` could be used instead here.

File: src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs [hunk 2]
```diff
@@ -185,70 +185,52 @@ public void TestEnvironmentProperty()
...
+                // but we start child process without them. Thus the set of variables from the child process could
+                // be a subset of variables from current process.)
+                var envEnv = new HashSet<string>(Environment.GetEnvironmentVariables().Cast<DictionaryEntry>().Select(e => e.Key + "=" + e.Value));
+                if (!envEnv.IsSupersetOf(actualEnv))
                 {
-                    currentProcEnv.Add(envVar.Key + "=" + envVar.Value);
+                    Assert.True(false, string.Format("Expected: {0}{1}Actual: {2}",
+                        string.Join(", ", envEnv.Except(actualEnv)),
+                        Environment.NewLine,
+                        string.Join(", ", actualEnv.Except(envEnv))));
```
Comment (akoeplinger) [inline src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs]: same here

File: src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs]: We added these to help diagnose when it fails.

**Comment by akoeplinger** [comment]
LGTM

> Something changed recently in the CI environment, such that Jenkins is now putting PR/commit/comment information in environment variables

FWIW, the ghprb Jenkins plugin always did this. I don't know why the newlines show up only now though.

File: src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs (hunk 1 above)
Comment (akoeplinger) [inline src/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs]: Ah yeah, I missed that this just logs the condition. We could still get rid of the string.Format by doing Assert.NotEqual on the strings, but I don't care much.

**Comment by bartonjs** [comment]
For some reason the Linux machine is trying to use Windows APIs?

```
00:33:48 Unhandled Exception: System.DllNotFoundException: Unable to load DLL 'api-ms-win-core-processenvironment-l1-1-0.dll': The specified module could not be found.
00:33:48  (Exception from HRESULT: 0x8007007E)
00:33:48    at Interop.mincore.GetCurrentDirectory(Int32 nBufferLength, StringBuilder lpBuffer)
00:33:48    at System.IO.Win32FileSystem.GetCurrentDirectory()
00:33:48    at Xunit.ConsoleClient.Program.Main(String[] args)
```

Did we somehow pull in an xUnit console program change that doesn't pass CI?

**Comment by stephentoub** [comment]
> Did we somehow pull in an xUnit console program change that doesn't pass CI?

@weshaggard?

**Comment by stephentoub** [comment]
@dotnet-bot test this please (maybe a random fluke?)

**Comment by weshaggard** [comment]
@dotnet-bot test this please

**Comment by Maxwe11** [comment]
LGTM

**Comment by mmitche** [comment]
@stephentoub What did you see?

**Comment by mmitche** [comment]
Hm....weird...I'll keep an eye on that.

**Pull request merged**